File: backend/src/main/kotlin/project/backend/core/domain/Document.kt
package project.backend.core.domain

import jakarta.persistence.*
import project.backend.core.enums.DocumentType
import project.backend.core.interfaces.entities.ISoftDelete
import java.sql.Date
import java.util.*

@Entity
class Document() : ISoftDelete() {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    lateinit var id: UUID

    @Column(nullable = true)
    @Enumerated(EnumType.STRING)
    var type: DocumentType? = null

    @Column
    lateinit var createdAt: Date

    @Column
    lateinit var storageUrl: String

    @ManyToOne(fetch = FetchType.LAZY)
    lateinit var folder: Folder

    constructor(createdAt: Date, storageUrl: String) : this() {
        this.createdAt = createdAt
        this.storageUrl = storageUrl
    }
}

File: backend/SDIA/Domain/Entities/Document.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Constants;
using Domain.Interfaces;

namespace Domain.Entities;

public class Document : IAudit, ISoftDelete
{
    [Key]
    public Guid Id { get; set; }
    
    [Required]
    public DocumentType DocumentType { get; set; }
    
    [Required]
    public string StorageUrl { get; set; } = null!;
    
    [ForeignKey("Folder")]
    public Guid FolderId { get; set; }
    
    public Folder Folder { get; set; } = null!;
    
    [Required]
    public DateTime CreatedAt { get; set; }

    [Required]
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
}